server/src/main/java/org/apache/iotdb/db/query/udf/core/reader/ConstantLayerPointReader.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.query.udf.core.reader;
+
+import org.apache.iotdb.db.exception.query.QueryProcessException;
+import org.apache.iotdb.db.query.expression.unary.ConstantExpression;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.utils.Binary;
+
+import org.apache.commons.lang3.Validate;
+
+import java.io.IOException;
+
+public class ConstantLayerPointReader implements LayerPointReader {
+
+  private final ConstantExpression expression;
+
+  public ConstantLayerPointReader(ConstantExpression expression) {
+    this.expression = Validate.notNull(expression);
+  }
+
+  @Override
+  public boolean isConstantPointReader() {
+    return true;
+  }
+
+  @Override
+  public boolean next() throws QueryProcessException, IOException {
+    return true;
+  }
+
+  @Override
+  public void readyForNext() {
+    // Do nothing
+  }
+
+  @Override
+  public TSDataType getDataType() {
+    return expression.getDataType();
+  }
+
+  @Override
+  public long currentTime() throws IOException {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public int currentInt() throws IOException {
+    if (TSDataType.INT32.equals(expression.getDataType())) {
+      return (int) expression.getValue();
```

Comment:
It's better to avoid the unboxing process, which will cause heavy performance issues. You can refer to org/apache/iotdb/db/query/udf/core/transformer/Transformer.java to find the solution. :D

Reply:
Good point. I will use the same way as the Transformer to avoid the performance issue.